src/lib.rs

```diff
@@ -309,6 +311,69 @@ where
     }
 }
 
+/// Compares two JSON values without panicking.
+///
+/// Returns a `Result` containing either `Ok(())` if the values match,
+/// or an `Err` with a [`Vec<DifferenceBuf>`](DifferenceBuf) describing the differences.
+///
+/// # Note:
+///
+/// This function performs some cloning and may be less efficient.
+///
+/// If you only need a string error message, use [`assert_json_matches_no_panic_to_string`] or the assertion macros.
+///
+/// # Examples
+///
+/// ```
+/// use serde_json_assert::{assert_json_matches_no_panic, Config, CompareMode};
+/// use serde_json::json;
+///
+/// let lhs = json!({ "a": 1, "b": 2 });
+/// let rhs = json!({ "a": 1 });
+/// let config = Config::new(CompareMode::Inclusive);
+///
+/// let result = assert_json_matches_no_panic(&lhs, &rhs, &config);
+/// assert!(result.is_ok());
+///
+/// let lhs = json!({ "a": 1 });
+/// let rhs = json!({ "a": 2 });
+/// let config = Config::new(CompareMode::Strict);
+///
+/// let result = assert_json_matches_no_panic(&lhs, &rhs, &config);
+/// assert!(result.is_err());
+/// ```
+pub fn assert_json_matches_no_panic<Lhs, Rhs>(
```

Comment:
Maybe we should drop the `assert_` prefix altogether? This isn't really an assert, because in my book an assert is supposed to panic. And giving it a new name isn't breaking. So maybe

```suggestion
pub fn json_matches<Lhs, Rhs>(
```

is better? And if you make `Difference` `Debug + Display`, it's a proper `Error`. I think this would make the old `assert_json_matches_no_panic` redundant, right?

Reply:
Or maybe something like `check_json_matches` or `test_json_matches` to convey that it's something that could fail, and not something that returns a boolean.

Reply:
I'm discussing with myself here, but after some thought I think I've landed on a good name.

`try_assert_json_matches`

I think the `try_assert_` prefix is compelling because it aligns well with Rust's existing conventions and indicates that it is a fallible version of the `assert_` macro.

Reply:
`try_assert_json_matches` sound good!

> I think this would make the old assert_json_matches_no_panic redundant, right?

The reason I wanted to keep it is for the speed, since the assert macros and the old `assert_json_matches_no_panic` itself are still using the borrow type `DifferenceRef<'_>` which should be faster than the new owned one.

Reply:
I just made a quick experiment, to figure out the overhead if we completely moved to the new `Difference` (So it like the Option 1 mentioned in https://github.com/hardselius/serde-json-assert/issues/1 )
test branch: https://github.com/B-2U/serde-json-assert/commit/1f5b781119a4473f05315837fdba3ee467089b79

When I ran `cargo test` on my pc (Intel i7 10700), all normal tests finished in 0.00 or 0.01s, the same as the original main branch, 
and for the slowest doc tests, the main branch took about 0.28 ~ 0.32s, and this option 1 branch took 0.29 ~ 0.35s, about 9% overhead in both worst case. 

In my use cases, I do not really care about the speed of json assertion, but maybe it will be useful for some other users?

Reply:
Another advantage of let the macros use the new method is it will be easier to impl for https://github.com/hardselius/serde-json-assert/issues/2
since it will get a real struct but not string

Reply:
```rust
use serde_json::json;
use serde_json_assert::try_assert_json_matches;
use serde_json_assert::Config;
use std::time::Instant;

#[test]
fn test_diff_json_values_speed() {
    let value1 = json!({
        "name": "Alice",
        "age": 30,
        "emails": ["alice@example.com", "alice@work.com"],
        "address": {
            "city": "Wonderland",
            "zip": "12345"
        }
    });

    let value2 = json!({
        "name": "Bob",
        "age": 25,
        "emails": ["bob@example.com"],
        "address": {
            "city": "Builderland",
            "zip": "54321"
        }
    });

    let iterations = 100_000;
    let start = Instant::now();
    let config = Config::new(serde_json_assert::CompareMode::Strict);

    for _ in 0..iterations {
        let _ = try_assert_json_matches(&value1, &value2, &config);
    }

    let duration = start.elapsed();
    println!(
        "Comparing two serde_json::Value {} times took: {:?}",
        iterations, duration
    );
}
```
I just make another test for `Option 1` on the branches `main` vs `make-diff-owned` ( directly making `Difference<'_>` a owned type ) 
The overhead of directly making `Difference<'_>` in main branch to own its data is around 5%

![image](https://github.com/user-attachments/assets/770c34b9-f79e-4ef4-b021-a8a2136a838f)
![image](https://github.com/user-attachments/assets/fcacc9f3-a971-48eb-bb55-b518f1b295e6)

Only 5% overhead is kind of attractive to me, since we wouldn't need 3 more new types and boilerplates, and it will be more friendly for #2.

Please let me know what you think, is option 1 a better idea for you?